homeassistant/components/alexa/capabilities.py

```diff
@@ -669,13 +668,10 @@ def get_property(self, name):
             current_level = self.entity.attributes.get(
                 media_player.ATTR_MEDIA_VOLUME_LEVEL
             )
-            try:
-                current = math.floor(int(current_level * 100))
-            except ZeroDivisionError:
-                current = 0
-            return current
```

Comment:
Please make sure that we always return an integer. Now we can return `None`. Also please include a test.

Reply:
Hi @balloob, I'm was not sure if that was the desired result. I thought that ZeroDivisionError catch was mistakenly placed for the value of zero. That logic will never happen because there is no division. The only exception I can see being thrown is TypeError which happens when current_level = None.

As for returning None, I believed that was appropriate based on the volume can be None when the media_player is off. I don't believe that is the same as volume=0.

I'll revise if you still think it's appropriate.

Reply:
It depends on what values Alexa supports. Do they support getting `None` for that field?

Reply:
No, but here is why I chose this path. Here is the trace of the specific issue:
```Error handling request
Traceback (most recent call last):
  File "/srv/homeassistant/lib/python3.7/site-packages/aiohttp/web_protocol.py", line 418, in start
    resp = await task
  File "/srv/homeassistant/lib/python3.7/site-packages/aiohttp/web_app.py", line 458, in _handle
    resp = await handler(request)
  File "/srv/homeassistant/lib/python3.7/site-packages/aiohttp/web_middlewares.py", line 119, in impl
    return await handler(request)
  File "/srv/homeassistant/lib/python3.7/site-packages/homeassistant/components/http/real_ip.py", line 39, in real_ip_middleware
    return await handler(request)
  File "/srv/homeassistant/lib/python3.7/site-packages/homeassistant/components/http/ban.py", line 72, in ban_middleware
    return await handler(request)
  File "/srv/homeassistant/lib/python3.7/site-packages/homeassistant/components/http/auth.py", line 135, in auth_middleware
    return await handler(request)
  File "/srv/homeassistant/lib/python3.7/site-packages/homeassistant/components/http/view.py", line 123, in handle
    result = await result
  File "/srv/homeassistant/lib/python3.7/site-packages/homeassistant/components/alexa/smart_home_http.py", line 120, in post
    hass, self.smart_home_config, message, context=core.Context(user_id=user.id)
  File "/srv/homeassistant/lib/python3.7/site-packages/homeassistant/components/alexa/smart_home.py", line 43, in async_handle_message
    response.merge_context_properties(directive.endpoint)
  File "/srv/homeassistant/lib/python3.7/site-packages/homeassistant/components/alexa/messages.py", line 189, in merge_context_properties
    for prop in endpoint.serialize_properties():
  File "/srv/homeassistant/lib/python3.7/site-packages/homeassistant/components/alexa/entities.py", line 264, in serialize_properties
    for prop in interface.serialize_properties():
  File "/srv/homeassistant/lib/python3.7/site-packages/homeassistant/components/alexa/capabilities.py", line 231, in serialize_properties
    prop_value = self.get_property(prop_name)
  File "/srv/homeassistant/lib/python3.7/site-packages/homeassistant/components/alexa/capabilities.py", line 674, in get_property
    current = math.floor(int(current_level * 100))
TypeError: unsupported operand type(s) for *: 'NoneType' and 'int'
```

Based on the trace, here is the function that calls the function I modified:
https://github.com/home-assistant/home-assistant/blob/deda2f86e7e138fd22f2853bc42ca72a6387976a/homeassistant/components/alexa/capabilities.py#L226-L238

If the function I have modified returns None, it will not attempt to package that property to send to Alexa. I thought this was most appropriate given we don't actually know what the volume is.

Reply:
The Alexa API supports an `int` between `0` and `100` for the volume property in this case.

But, `None` is acceptable here. When the properties are serialized for the API response it checks for `None` and does not include the `context.properties` object representing that property. This is how `STATE_UNAVAILABLE` and `STATE_UNKNOWN` are handled too for some entities. This also allows for the `Alexa.EndpointHealth` Interface to report to the Alexa API connectivity status without a potentially unnecessary `Alexa.ErrorResponse`.

Reply:
Ok. In that case this PR is ok to merge when you make sure you return `None` instead of changing it to an `elif` (so it's more explicit) and add a test 👍

Reply:
I removed the elif -- the function will always explicitly return a valid value. 
I added a test for all valid volume values and verified it executes the code changed.